servlet/src/main/java/org/togglz/servlet/activation/ClientIpActivationStrategy.java

```diff
@@ -39,28 +45,49 @@ public String getName()
    @Override
    public boolean isActive(FeatureState featureState, FeatureUser user)
    {
-
       HttpServletRequest request = HttpServletRequestHolder.get();
       if (request != null) {
 
-         String allowedIpsParam = featureState.getParameter(PARAM_IPS);
-         List<String> allowsIps = Strings.splitAndTrim(allowedIpsParam, "[\\s,]+");
-
-         // TODO: This should support a simple form of subnet matching
-         return allowsIps.contains(request.getRemoteAddr());
-
+         List<String> parts = Strings.splitAndTrim(featureState.getParameter(PARAM_IPS), "[\\s,]+");
+
+         List<InetAddress> ips = new ArrayList<>();
+         List<CIDRUtils> cidrUtils = new ArrayList<>();
+         for (String part : parts) {
+            try {
+               if (part.contains("/")) {
+                  cidrUtils.add(new CIDRUtils(part));
+               } else {
+                  ips.add(InetAddress.getByName(part));
+               }
+            } catch (Exception e) {
+               log.warn("Ignoring illegal IP address or CIDR range " + part);
```

Comment:
First you should catch `UnknownHostException` instead of `Exception` here.

And we should perhaps use the `ParameterBuilder` to create a validation rule for parameter values. The builder supports this using regular expressions. But you could also provider you own `Parameter` implementation which provides custom validation code. See `ScriptEngineActivationStrategy` for an example. This way we could catch invalid strategy parameters before they are saved.